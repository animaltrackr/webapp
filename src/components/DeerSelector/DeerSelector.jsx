import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Select } from 'antd';

const { Option } = Select;

class DeerSelector extends Component {
	render() {
		const { deerStates, toggleDeer, loading } = this.props;
		const selectedDeer = deerStates
			.filter(deer => deer.visible)
			.map(deer => deer.id);

		return (
			<Select
				mode="multiple"
				style={{ width: '100%' }}
				placeholder="Deer to Display"
				value={selectedDeer}
				onSelect={toggleDeer}
				onDeselect={toggleDeer}
				disabled={loading}
			>
				{this.props.deerStates
					? this.props.deerStates.map(deer => {
							return <Option key={deer.id}>{deer.name}</Option>;
					  })
					: null}
			</Select>
		);
	}
}

DeerSelector.propTypes = {
	deerStates: PropTypes.arrayOf(PropTypes.object).isRequired,
	toggleDeer: PropTypes.func.isRequired,
	loading: PropTypes.bool.isRequired,
};

export default DeerSelector;
