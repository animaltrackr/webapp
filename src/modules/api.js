const SERVER_URL = `http://api.animaltrackr.com`;

const METHOD = {
	GET: 'GET',
	POST: 'POST',
	PUT: 'PUT',
	DELETE: 'DELETE',
};

// Log outgoing requests
const logRequest = (method, url, body = undefined) => {
	let message = `${method} Request made to ${url}`;
	if (body) message += `\nWith body: ${JSON.stringify(body)}`;

	console.info(message);
};

// Log incoming responses
const logResponse = (method, url, body = undefined) => {
	let message = `${method} Response from ${url}`;
	if (body) message += `\nWith body: ${JSON.stringify(body)}`;

	console.info(message);
};

// Compose fetch with logging & JSON parsing
const myFetch = (url, options = {}) => {
	const method = options.method || METHOD.GET;
	logRequest(method, url, options);

	return fetch(url, {
		headers: {
			// 'Content-Type': 'application/json',
			...(options.headers || {}),
		},
		...options,
	})
		.then(response => response.json())
		.then(resBody => {
			logResponse(method, url, resBody);
			return resBody;
		});
};

export const createTracker = data => {
	const url = `${SERVER_URL}/animal/trackers`;
	return myFetch(url, {
		method: METHOD.POST,
		body: JSON.stringify(data),
	});
};

export const readTrackers = () => {
	const url = `${SERVER_URL}/animal/trackers`;
	return myFetch(url);
};

export const readTracker = trackerId => {
	const url = `${SERVER_URL}/animal/trackers/${trackerId}`;
	return myFetch(url);
};

export const updateTracker = (trackerId, data) => {
	const url = `${SERVER_URL}/animal/trackers/${trackerId}`;
	return myFetch(url, {
		method: METHOD.PUT,
		body: JSON.stringify(data),
	});
};

export const deleteTracker = trackerId => {
	const url = `${SERVER_URL}/animal/trackers/${trackerId}`;
	return myFetch(url, {
		method: METHOD.DELETE,
	});
};

export const createPoint = data => {
	const url = `${SERVER_URL}/animal/points`;
	return myFetch(url, {
		method: METHOD.POST,
		body: JSON.stringify(data),
	});
};

export const readPoints = () => {
	const url = `${SERVER_URL}/animal/points`;
	return myFetch(url);
};

export const readPoint = pointId => {
	const url = `${SERVER_URL}/animal/points/${pointId}`;
	return myFetch(url);
};

export const updatePoint = (pointId, data) => {
	const url = `${SERVER_URL}/animal/points/${pointId}`;
	return myFetch(url, {
		method: METHOD.PUT,
		body: JSON.stringify(data),
	});
};

export const deletePoint = pointId => {
	const url = `${SERVER_URL}/animal/points/${pointId}`;
	return myFetch(url, {
		method: METHOD.DELETE,
	});
};
