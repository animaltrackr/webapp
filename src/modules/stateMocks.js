// eslint-disable-next-line import/prefer-default-export
export const deerStates = [
	{
		name: 'ohwowidzz',
		visible: false,
		colour: 'blue',
		id: 'bc6bf7b3-b173-4355-b5d2-ac1cdb2263ad',
		status: 'A',
		max_error_radius: '20.0',
		location_method: 'B',
		points: [
			{
				id: 'a4ce4e6b-9fdc-4311-ad6a-5c67985c2e22',
				tracker: 'bc6bf7b3-b173-4355-b5d2-ac1cdb2263ad',
				timestamp: '2019-07-23T06:40:15.407000Z',
				geo_lat: '48.463325',
				geo_long: '-123.311751',
				geo_error_radius: '1.0',
				geo_method: 'G',
			},
			{
				id: '49b497aa-461f-40dc-9ba0-774d518f2354',
				tracker: 'bc6bf7b3-b173-4355-b5d2-ac1cdb2263ad',
				timestamp: '2019-07-23T06:40:31.092000Z',
				geo_lat: '48.463325',
				geo_long: '-123.311751',
				geo_error_radius: '1.0',
				geo_method: 'G',
			},
			{
				id: 'bdf5812c-d44c-4071-b768-8e81e14e3526',
				tracker: 'bc6bf7b3-b173-4355-b5d2-ac1cdb2263ad',
				timestamp: '2019-07-23T06:40:32.208000Z',
				geo_lat: '48.463325',
				geo_long: '-123.311751',
				geo_error_radius: '1.0',
				geo_method: 'G',
			},
			{
				id: '74d87fa9-d6c2-4523-98b6-69c257b2792e',
				tracker: 'bc6bf7b3-b173-4355-b5d2-ac1cdb2263ad',
				timestamp: '2019-07-24T05:29:07Z',
				geo_lat: '48.424610',
				geo_long: '-123.342901',
				geo_error_radius: '3.0',
				geo_method: 'L',
			},
			{
				id: '51c528ff-1f9f-4a3c-b9ae-47b2c790cb1b',
				tracker: 'bc6bf7b3-b173-4355-b5d2-ac1cdb2263ad',
				timestamp: '2019-07-24T05:29:07.010000Z',
				geo_lat: '48.424610',
				geo_long: '-123.342901',
				geo_error_radius: '3.0',
				geo_method: 'L',
			},
			{
				id: 'd9d18be5-6fd1-427f-b5aa-2f7e80e9f8c9',
				tracker: 'bc6bf7b3-b173-4355-b5d2-ac1cdb2263ad',
				timestamp: '2019-07-24T05:29:07Z',
				geo_lat: '48.424610',
				geo_long: '-123.342901',
				geo_error_radius: '3.0',
				geo_method: 'L',
			},
			{
				id: '65541dcd-dc46-424c-b14a-e58f8f79e6b0',
				tracker: 'bc6bf7b3-b173-4355-b5d2-ac1cdb2263ad',
				timestamp: '2019-07-24T05:58:03Z',
				geo_lat: '48.424714',
				geo_long: '-123.342991',
				geo_error_radius: '3.0',
				geo_method: 'L',
			},
			{
				id: '49ea0496-4f0a-413c-a0d5-91de9a6b77f9',
				tracker: 'bc6bf7b3-b173-4355-b5d2-ac1cdb2263ad',
				timestamp: '2019-07-24T05:58:33Z',
				geo_lat: '48.424743',
				geo_long: '-123.342952',
				geo_error_radius: '4.0',
				geo_method: 'L',
			},
		],
	},
	{
		name: 'deeeer',
		visible: false,
		colour: 'white',
		id: 'b56e6345-b434-4f7f-9d1c-559e2d2aecc9',
		status: 'D',
		max_error_radius: '2512.0',
		location_method: 'B',
		points: [
			{
				id: '08f78222-877b-499e-83a2-dbd148ded657',
				tracker: 'b56e6345-b434-4f7f-9d1c-559e2d2aecc9',
				timestamp: '2019-07-23T07:50:58Z',
				geo_lat: '120.500000',
				geo_long: '45.200000',
				geo_error_radius: '10.0',
				geo_method: 'L',
			},
			{
				id: '0b4c7f10-8f7f-443d-bc4a-5ac7047c8f34',
				tracker: 'b56e6345-b434-4f7f-9d1c-559e2d2aecc9',
				timestamp: '2019-07-23T07:50:58Z',
				geo_lat: '120.500000',
				geo_long: '45.200000',
				geo_error_radius: '10.0',
				geo_method: 'L',
			},
			{
				id: '992c3cce-2918-42f5-ad6c-e85e3306a146',
				tracker: 'b56e6345-b434-4f7f-9d1c-559e2d2aecc9',
				timestamp: '2019-07-23T07:50:58Z',
				geo_lat: '120.500000',
				geo_long: '45.200000',
				geo_error_radius: '10.0',
				geo_method: 'L',
			},
		],
	},
];
