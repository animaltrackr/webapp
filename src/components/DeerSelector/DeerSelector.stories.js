import React from 'react';
import { storiesOf } from '@storybook/react';
import DeerSelector from './DeerSelector';

storiesOf('DeerSelector', module).add('DeerSelector', () => (
	<div>
		<DeerSelector />
	</div>
));
