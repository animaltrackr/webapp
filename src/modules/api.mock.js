import { deerStates } from 'modules/stateMocks';

export default function() {
	// eslint-disable-next-line global-require
	const api = require('modules/api');
	jest.mock('modules/api');

	api.createTracker.mockResolvedValue(new Promise(res => res({})));
	api.readTrackers.mockResolvedValue(new Promise(res => res(deerStates)));
	api.readTracker.mockResolvedValue(new Promise(res => res({})));
	api.updateTracker.mockResolvedValue(new Promise(res => res({})));
	api.deleteTracker.mockResolvedValue(new Promise(res => res({})));
	api.createPoint.mockResolvedValue(new Promise(res => res({})));
	api.readPoints.mockResolvedValue(new Promise(res => res([])));
	api.readPoint.mockResolvedValue(new Promise(res => res({})));
	api.updatePoint.mockResolvedValue(new Promise(res => res({})));
	api.deletePoint.mockResolvedValue(new Promise(res => res({})));
}
