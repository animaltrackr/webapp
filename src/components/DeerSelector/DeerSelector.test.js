import React from 'react';
import { shallow } from 'enzyme';
import { deerStates } from 'modules/stateMocks';
import DeerSelector from './DeerSelector';

describe('<DeerSelector />', () => {
	test('renders', () => {
		const wrapper = shallow(<DeerSelector deerStates={deerStates} />);
		expect(wrapper).toMatchSnapshot();
	});
});
