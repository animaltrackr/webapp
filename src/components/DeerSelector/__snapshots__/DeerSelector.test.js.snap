// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeerSelector /> renders 1`] = `ShallowWrapper {}`;
